### ISSUE TITLE
Q&A

### ISSUE THREAD

**Issue by thenaim** [thenaim/ionic-tk-social-network#9]

**Comment by SnowBases** [comment]
Hello @thenaim. This template app is really cool.
I have some question about components,
why you didnt make a component module instead?

**Comment by thenaim** [comment]
@SnowBases, The main reason is the lack of free time (since I work) to implement this approach, but in the code I try to separate templates into components and modules (angular).

**Comment by nguoianphu** [comment]
Hello @thenaim ,
Thank you so much for a great app. 👍 
Could you please tell me how to run it and log in? I can build and run it by ```npm run start```, but I don't know how to log in or register. Is there any demo account to use?

**Comment by thenaim** [comment]
@nguoianphu, you can use any valid email addres and password min 6 digits.

**Comment by nguoianphu** [comment]
Thanks @thenaim, got it. I'm able to build & run it on web. I also can build it on Github action & generate APKs to Release. However, the app seems slow when running on Android...

**Comment by thenaim** [comment]
@nguoianphu, Did you compile the production version of angular (ionic), and also compile for android (apk)? If not, then the application will be slow to work, as many plugins and helper functions in the dev version are running under the hood. 

And also the application requires a lot of optimization, so if there is a desire to work on this, then welcome)

**Comment by nguoianphu** [comment]
Yes, I've compiled it in production mode. Please check my Github action. BTW, I'm learning React & Ionic so I'm not interested in Angular. :) 
https://github.com/nguoianphu/ionic-tk-social-network/releases

**Comment by thenaim** [comment]
@nguoianphu,  I just checked the apk on a real device. It really works slowly, but as I wrote above, it is necessary to optimize the interaction with the server. Such as pagination, preloading modules, requesting the server after full rendering, etc. First of all, I will deal with this. Thank you.

I see you worked with github actions, do you want to work and do PR, where all this will be set up for this project? To configure action for all PR`s for main branch.

**Comment by nguoianphu** [comment]
@thenaim , just see you added actions to deploy the web version automatically. It's great 👍 !

But it looks like we deploy the web version from main branch, instead of web. Does it make sense?

**Comment by thenaim** [comment]
@nguoianphu, Thank you. This is done to avoid committing www changes to the main branch every time and the web version is always separated from the main branch. Or do you have an idea?

**Comment by nguoianphu** [comment]
Oh, I was wrong -_-. Never mind, sorry.

**Comment by thenaim** [comment]
@nguoianphu, It's okay, I thought about this task for a long time)

**Comment by thenaim** [comment]
@nguoianphu, I have a question for you, how we can optimize github actions? Can we somehow separate them, or into some stages? Any ideas?

**Comment by nguoianphu** [comment]
@thenaim , exactly what I want to do.
Please refer this, I like it and plan to apply: https://github.com/zyrouge/yukino-app/actions

**Comment by thenaim** [comment]
@nguoianphu, Yes, this approach is very good. for example, first of all, run the code check (eslint), if no errors, then the rest of the actions. I think you already understood the concept.

**Comment by nguoianphu** [comment]
@thenaim , can you register this repo to https://www.codefactor.io? Then we can have badge for it. If it's ok, I will open a PR for this [commit](https://github.com/nguoianphu/ionic-tk-social-network/commit/3f2b60cb87ead78e93eeb2717b0652c0916658b4).

**Comment by thenaim** [comment]
Hi, @nguoianphu, good idea. Yes, just now registered.
